projects/packages/search/src/instant-search/class-instant-search.php

```diff
@@ -309,97 +311,161 @@ public function get_search_aggregations_results() {
 	 * @since  8.3.0
 	 */
 	public function auto_config_search() {
-		// Set default result format to "expanded".
-		update_option( Options::OPTION_PREFIX . 'result_format', Options::RESULT_FORMAT_EXPANDED );
-
 		$this->auto_config_excluded_post_types();
 		$this->auto_config_overlay_sidebar_widgets();
-		$this->auto_config_woo_result_format();
+		$this->auto_config_theme_sidebar_search_widget();
+		$this->auto_config_result_format();
 	}
 
 	/**
-	 * Automatically copy configured search widgets into the overlay sidebar
+	 * Automatically copy configured search widgets from theme sidebar to the overlay sidebar.
+	 * If there's nothing to copy, we create one.
 	 *
 	 * @since  8.8.0
 	 */
 	public function auto_config_overlay_sidebar_widgets() {
-		global $wp_registered_sidebars;
 		$sidebars = get_option( 'sidebars_widgets', array() );
-		$slug     = Helper::FILTER_WIDGET_BASE;
 
+		// If there's JP search widget in overly sidebar, skip.
 		if ( isset( $sidebars['jetpack-instant-search-sidebar'] ) ) {
 			foreach ( (array) $sidebars['jetpack-instant-search-sidebar'] as $widget_id ) {
-				if ( 0 === strpos( $widget_id, $slug ) ) {
+				if ( 0 === strpos( $widget_id, Helper::FILTER_WIDGET_BASE ) ) {
 					// Already configured.
 					return;
 				}
 			}
 		}
```

Comment:
Could we replace this by checking if `$sidebar_jp_searchbox_wiget_id` (L349) is false?

Reply:
Addressed in 3839f610c1